capsules/src/process_console.rs

```diff
@@ -291,11 +291,9 @@ impl Default for Command {
 
 impl PartialEq<[u8; COMMAND_BUF_LEN]> for Command {
     fn eq(&self, other_buf: &[u8; COMMAND_BUF_LEN]) -> bool {
-        self.buf
-            .iter()
-            .zip(other_buf.iter())
-            .take_while(|(a, b)| **a != TERMINATOR || **b != TERMINATOR)
-            .all(|(a, b)| a == b)
+        let mut bytes_iter = self.buf.iter().zip(other_buf.iter());
+        bytes_iter.all(|(a, b)| *a == *b && *a != TERMINATOR);
+        bytes_iter.next().eq(&Some((&TERMINATOR, &TERMINATOR)))
```

Comment:
Why is this here?

Reply:
> Why is this here?

Write above why this would work

Reply:
This works https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b3a02363bd56e07752d43cd6a6a065ed

Reply:
Yes but doesn't it compare the same elements for two times? First it compares in the take_while and then in the all, while the upper implementation compares the elements just one time

Reply:
Done

Reply:
Thanks, take a look at the this https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a58d1d8003d8960252e7c20da76a12d6, your code skips one item (the one that does not match) and returns true.